aptible/app.go

```diff
@@ -0,0 +1,78 @@
+package aptible
+
+import (
+	"fmt"
+
+	"github.com/go-openapi/runtime"
+	deploy "github.com/reggregory/go-deploy/client"
+	"github.com/reggregory/go-deploy/client/operations"
+	"github.com/reggregory/go-deploy/models"
+)
+
+func CreateApp(client *deploy.DeployAPIV1, token runtime.ClientAuthInfoWriter, handle string, account_id int64) (*models.InlineResponse2011, error) {
```

Comment:
We'll want these to be methods of our own struct. 

So like:

```
type Client struct {
     generatedClient *deploy.DeployAPIV1
     token runtime.ClientAuthInfoWriter
    // Maybe config or whatever else
}

func (client *Client) CreateApp (handle string, account_id int64) ...
```

The idea being to stash all the repetitive stuff into our struct and then it gets passed in as the method receiver so the user only has to care about the params that really matter to them.

Make sense?

You can do that now in this PR or maybe wait until you do one more resource, just to be sure of the repetitive parts of the design. I'll leave that choice up to you.

Reply:
Would we initialize this Client struct in the ConfigureFunc of the Provider and then pass it into the required methods as a receiver (accessed via the m interface{} argument in Terraform CRUD funcs)?

Reply:
Yeah, that'd be the idea. To have one thing that encapsulates all the stuff that a user needs to talk to our API.